apps/src/lib/levelbuilder/CourseVersionPublishingEditor.jsx

```diff
@@ -54,6 +55,34 @@ export default class CourseVersionPublishingEditor extends Component {
     this.props.updateFamilyName(e.target.value);
   };
 
+  /*
+   * We only want a course to be able to progress
+   * forward in published states. So as the editor updates the
+   * published start of the course the options to update in the
+   * future should get smaller.
+   */
+  getAvailablePublishedStates = currentState => {
+    const availablePublishedStates = {
+      in_development: [
+        PublishedState.in_development,
+        PublishedState.pilot,
+        PublishedState.beta,
+        PublishedState.preview,
+        PublishedState.stable
+      ],
+      pilot: [PublishedState.pilot],
```

Comment:
interesting, I like this restriction. I wonder though if we should warn curriculum writers if they change into this published state (or change out of in_development to any other state) what exactly (or generally) is getting locked down when they do so? we could use a simple `window.confirm` prompt.

Reply:
Added!